src/query/generated/proto/rpcpb/query.proto

```diff
@@ -56,19 +56,29 @@ enum MatcherType {
 
 message FetchOptions {
 	int64 limit                      = 1;
-	RestrictFetchOptions restrict    = 2;
+	RestrictQueryOptions restrict    = 2;
 	int64 lookbackDuration           = 3;
 	FanoutOption unaggregated        = 4;
 	FanoutOption aggregated          = 5;
 	FanoutOption aggregatedOptimized = 6;
 	bool includeResolution           = 7;
 }
 
-message RestrictFetchOptions {
+message RestrictQueryOptions {
+	RestrictFetchType RestrictFetchType = 3;
```

Comment:
nit: should we change these to `Query*` as well? Since we changed the name of the outer message from `Fetch` -> `Query`.